### PR TITLE
Added request management

### PR DESCRIPTION
**Pull request by schaefi**
Added support for obtaining request status and also waiting
for request completion for service management tasks

### CODE REVIEW

**Comment by schaefi** [comment]
> Is --id really a command?

no, actually a good catch we have this mix for all commands. I have separated it in the usage message now for all of them

**Comment by schaefi** [comment]
> Maybe also capitalize 'log'?

don't want to do this now. the use of the global log instance is a statement of the form

```python
from logger import log
```

I agree because the log instance is application global we could change this to LOG. However I would consider this an extra refactoring task which we should not do now

ok ?

File: azurectl/azure_account.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def publishsettings(self):
         )
         return result
 
-    def __get_service(self):
+    def get_service(self):
         if self.service:
             return
```
Comment (rjschwei) [inline azurectl/azure_account.py]: Based on the name of the method I would expect

return self.service

at this point??

File: azurectl/compute_request_task.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        self.result = DataCollector()
+        self.out = DataOutput(
+            self.result,
+            self.global_args['--output-format'],
+            self.global_args['--output-style']
+        )
+
+        self.account = AzureAccount(self.config)
+        self.account.get_service()
```
Comment (rjschwei) [inline azurectl/compute_request_task.py]: A "get_" method should return something. Whatever it returns should then be used.

File: azurectl/compute_request_task.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                'request:' + format(self.command_args['--id']),
+                format(self.__get_status())
+            )
+            self.out.display()
+        elif self.command_args['wait']:
+            self.__wait_for_request_to_complete()
+
+    def __get_status(self):
+        return self.request_result.status(
+            self.account.service
```
Comment (rjschwei) [inline azurectl/compute_request_task.py]: Based on the name I would expect a status indicator to be returned, True, False, Pending, Busy, Completed as some status indicators that I can think of. But the method appears to return a service, which from my perspective is not a status.

**Comment by rjschwei** [comment]
On 09/23/2015 06:14 AM, Marcus Schäfer wrote:
>     Maybe also capitalize 'log'?
> 
> don't want to do this now. the use of the global log instance is a
> statement of the form
> 
> from logger import log
> 
> I agree because the log instance is application global we could change
> this to LOG. However I would consider this an extra refactoring task
> which we should not do now
> 
> ok ?

It should not be LOG, all caps is intended for constants. A singleton is
not a constant.



-- 
Robert Schweikert                   MAY THE SOURCE BE WITH YOU
Public Cloud Architect                         LINUX
rjschwei@suse.com
IRC: robjo

File: azurectl/compute_request_task.py (hunk 3 above)
Comment (schaefi) [inline azurectl/compute_request_task.py]: ? 
return self.request_result.status(...)

it does what you wrote, it returns the status

File: azurectl/compute_request_task.py (hunk 2 above)
Comment (schaefi) [inline azurectl/compute_request_task.py]: yes this is misleading I'm going to change it

File: azurectl/version.py [hunk 4]
```diff
@@ -14,4 +14,4 @@
 """
     Global version information used in azurectl and the package
 """
-__VERSION__ = '0.11.3'
+__VERSION__ = '0.12.3'
```
Comment (bear454) [inline azurectl/version.py]: Bumping the minor version should reset the subversion: `'0.12.0'`, please.

**Pull request merged**